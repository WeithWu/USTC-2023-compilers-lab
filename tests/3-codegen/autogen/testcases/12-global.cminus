int seed;

int randomLCG(void) {
    seed = seed * 1103515245 + 12345;
    return seed;
}

int randBin(void) {
    if (randomLCG() > 0)
        return 1;
    else
        return 0;
}

/* random walk */
int returnToZeroSteps(void) {
    int x;
    int steps;

    x = 0;
    steps = 0;

    while (steps < 20) {
        if (randBin())
            x = x + 1;
        else
            x = x - 1;

        steps = steps + 1;
        if (x == 0)
            return steps;
    }
    return 20;
}

int main(void) {
    int i;
    i = 0;

    seed = 3407;

    while (i < 20) {
        output(returnToZeroSteps());
        i = i + 1;
    }
    return 0;
}
