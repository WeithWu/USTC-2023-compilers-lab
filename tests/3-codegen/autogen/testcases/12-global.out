4
2
2
4
8
2
2
2
2
2
6
2
10
8
4
2
20
2
2
8
0
